reverse_engineering/databaseService/databaseService.js

```diff
@@ -316,152 +425,276 @@ const getDatabaseMemoryOptimizedTables = async (connectionClient, dbName, logger
+			containerName: '',
+			entityName: '',
+		});
+
+		return [];
+	}
 };
 
-const getTableDefaultConstraintNames = async (connectionClient, dbName, tableName, schemaName) => {
+const getTableDefaultConstraintNames = async ({ connectionClient, dbName, tableName, schemaName, logger }) => {
+	logger.log('info', { message: `Get '${tableName}' table default constraint names.` }, 'Reverse Engineering');
 	const currentDbConnectionClient = await getNewConnectionClientByDb(connectionClient, dbName);
-	return mapResponse(currentDbConnectionClient.query`
-	SELECT
-		ac.name as columnName,
-		dc.name
-	FROM 
-		sys.columns as ac
-			INNER JOIN
-		sys.tables
-			ON ac.object_id = tables.object_id
-			INNER JOIN 
-		sys.schemas
-			ON tables.schema_id = schemas.schema_id
-			INNER JOIN
-		sys.default_constraints as dc
-			ON ac.default_object_id = dc.object_id
-	WHERE 
-			schemas.name = ${schemaName}
-		AND tables.name = ${tableName}
-	`);
+
+	try {
+		return mapResponse(
+			await currentDbConnectionClient.query`
+			SELECT
+				ac.name as columnName,
+				dc.name
+			FROM 
+				sys.columns as ac
+					INNER JOIN
+				sys.tables
+					ON ac.object_id = tables.object_id
+					INNER JOIN 
+				sys.schemas
+					ON tables.schema_id = schemas.schema_id
+					INNER JOIN
+				sys.default_constraints as dc
+					ON ac.default_object_id = dc.object_id
+			WHERE 
+					schemas.name = ${schemaName}
+				AND tables.name = ${tableName}
+			`,
+		);
+	} catch (error) {
+		logger.log(
+			'error',
+			{ message: error.message, stack: error.stack, error },
+			`Get '${schemaName}.${tableName}' table default constraint names`,
+		);
+		logger.progress({
+			message: `Warning: failed to reverse-engineer ${tableName} table default constraint names.`,
+			containerName: '',
+			entityName: tableName,
+		});
+
+		return [];
+	}
 };
 
-const getDatabaseUserDefinedTypes = async (connectionClient, dbName, logger) => {
+const getDatabaseUserDefinedTypes = async ({ connectionClient, dbName, logger }) => {
 	const currentDbConnectionClient = await getNewConnectionClientByDb(connectionClient, dbName);
 
 	logger.log('info', { message: `Get '${dbName}' database UDTs.` }, 'Reverse Engineering');
 
-	return mapResponse(currentDbConnectionClient.query`
-		select * from sys.types
-		where is_user_defined = 1
-	`);
+	try {
+		return mapResponse(
+			await currentDbConnectionClient.query`
+			select * from sys.types
+			where is_user_defined = 1
+		`,
+		);
+	} catch (error) {
+		logger.log(
+			'error',
+			{ message: error.message, stack: error.stack, error },
+			`Retrieve '${dbName}' database UDTs`,
+		);
+		logger.progress({
+			message: `Warning: failed to reverse-engineer ${dbName} database UDTs.`,
+			containerName: '',
+			entityName: '',
+		});
+
+		return [];
+	}
 };
 
-const mapResponse = async (response = {}) => {
-	return (await response).recordset;
+const mapResponse = (response = {}) => {
+	return response?.recordset;
```

Comment:
Are you sure that `response` can never be a promise instead of an actual response?

Reply:
I ensured that by applying ```await``` for every query result we use in databaseService.js